src/test/java/net/cryptic_game/microservice/network/model/NetworkTest.java

```diff
@@ -0,0 +1,303 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import javax.persistence.NoResultException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({MicroService.class, Database.class})
+public class NetworkTest {
+
+    private Random rand;
+    private List<Network> networks;
+
+    @Mock
+    private MicroService microService;
+
+    @Mock
+    private Database database;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private Session session;
+
+    @Mock
+    private CriteriaBuilder criteriaBuilder;
+
+    @Mock
+    private CriteriaQuery<Network> criteriaQuery;
+
+    @Mock
+    private Root<Network> root;
+
+    @Mock
+    private Query<Network> typedQuery;
+
+    @Before
+    public void setUp() {
+        rand = new Random();
+        networks = new ArrayList<>();
+
+        PowerMockito.mockStatic(Database.class);
+        PowerMockito.when(Database.getInstance()).thenReturn(database);
+        PowerMockito.when(database.openSession()).thenReturn(session);
+        PowerMockito.when(session.getTransaction()).thenReturn(transaction);
+
+        PowerMockito.when(session.save(Mockito.any(Network.class))).thenAnswer((InvocationOnMock invocationOnMock) -> {
+            networks.add(invocationOnMock.getArgument(0));
+            return null;
+        });
+
+        PowerMockito.when(session.get(Mockito.eq(Network.class), Mockito.any(UUID.class))).then((InvocationOnMock invocationOnMock) -> {
+            UUID network = invocationOnMock.getArgument(1);
+
+            for (Network all : networks) {
+                if(all.getUUID().equals(network)) {
+                    return all;
+                }
+            }
+            return null;
+        });
+
+        PowerMockito.when(session.getCriteriaBuilder()).thenReturn(criteriaBuilder);
+        PowerMockito.when(criteriaBuilder.createQuery(Network.class)).thenReturn(criteriaQuery);
+        PowerMockito.when(criteriaQuery.from(Network.class)).thenReturn(root);
+        PowerMockito.when(session.createQuery(criteriaQuery)).thenReturn(typedQuery);
+
+        PowerMockito.mockStatic(MicroService.class);
+        PowerMockito.when(MicroService.getInstance()).thenReturn(microService);
+        PowerMockito.when(microService.contactMicroService(Mockito.eq("device"), Mockito.eq(new String[]{"owner"}), Mockito.any()))
+                .thenReturn(JSONBuilder.anJSON().add("owner", UUID.randomUUID()).build());
+    }
+
+    @Test
+    public void constructorTest() {
```

Comment:
This is no constructor test!
You are testing the getters.

Reply:
I'm testing both